api-admin/src/main/java/bio/terra/pearl/api/admin/service/forms/SurveyExtService.java

```diff
@@ -58,6 +64,59 @@ public Survey create(String portalShortcode, Survey survey, AdminUser adminUser)
     return surveyService.create(survey);
   }
 
+  public void delete(String portalShortcode, String surveyStableId, AdminUser adminUser) {
+    Portal portal = authUtilService.authUserToPortal(adminUser, portalShortcode);
+    List<Survey> existingVersions = surveyService.findByStableId(surveyStableId);
+    List<UUID> existingVersionIds = existingVersions.stream().map(BaseEntity::getId).toList();
+
+    if (existingVersionIds.size() == 0) {
+      throw new NotFoundException("Survey not found in portal " + portal.getName());
+    }
+
+    // Find all of the configured surveys that use any version of the specified survey
+    List<StudyEnvironmentSurvey> referencingSurveys =
+        existingVersionIds.stream()
+            .flatMap(
+                surveyId -> {
+                  List<StudyEnvironmentSurvey> configuredSurveys =
+                      studyEnvironmentSurveyService.findAllStudyEnvsWithSurveyId(surveyId);
+                  return configuredSurveys.stream();
+                })
+            .toList();
+
+    // Resolve all of the study environments that contain the configured surveys
+    List<StudyEnvironment> referencingStudyEnvs =
+        referencingSurveys.stream()
+            .map(
+                configuredSurvey -> {
+                  return studyEnvironmentService
+                      .find(configuredSurvey.getStudyEnvironmentId())
+                      .get();
+                })
+            .toList();
+
+    if (referencingStudyEnvs.stream()
+        .anyMatch(r -> !EnvironmentName.sandbox.equals(r.getEnvironmentName()))) {
+      throw new IllegalArgumentException(
+          "Survey is referenced by a non-sandbox environment and cannot be deleted");
+    } else {
+      // At this point, only sandbox environments reference the survey, so we can unlink
+      // the survey from each env to prepare for full deletion.
+      referencingSurveys.forEach(
+          configuredSurvey -> {
+            studyEnvironmentSurveyService.delete(
+                configuredSurvey.getId(), CascadeProperty.EMPTY_SET);
+          });
+
+      // Delete the survey
+      for (Survey survey : existingVersions) {
+        if (portal.getId().equals(survey.getPortalId())) {
```

Comment:
I think this check should be moved up to the beginning -- the original list of referencingSurveys should be filtered by the Portal.  As you have it, if a version of the survey is owned by another portal, it will be detached but not deleted, which I don't think is what we want